src/meta/server_state.cpp

```diff
@@ -3736,5 +3734,197 @@ void server_state::update_partition_max_replica_count_locally(
              new_config_str);
 }
 
+// ThreadPool: THREAD_POOL_META_SERVER
+void server_state::recover_from_max_replica_count_env()
+{
+    std::vector<std::pair<std::shared_ptr<app_state>, int32_t>> tasks;
+    {
+        zauto_read_lock l(_lock);
+        for (auto &e : _exist_apps) {
+            auto &app = e.second;
+            if (app->status != app_status::AS_AVAILABLE) {
+                continue;
+            }
+
+            auto iter = app->envs.find(replica_envs::UPDATE_MAX_REPLICA_COUNT);
+            if (iter == app->envs.end()) {
+                continue;
+            }
+
+            std::vector<std::string> args;
+            utils::split_args(iter->second.c_str(), args, ';');
+            if (args[0] != "updating") {
+                continue;
+            }
+
+            int32_t max_replica_count = 0;
+            if (args.size() < 2 || !dsn::buf2int32(args[1], max_replica_count) ||
+                max_replica_count <= 0) {
+                dassert_f(false,
+                          "invalid max_replica_count_env: app_name={}, app_id={}, "
+                          "max_replica_count={}, {}={}",
+                          app->app_name,
+                          app->app_id,
+                          app->max_replica_count,
+                          iter->first,
+                          iter->second);
+            }
+
+            tasks.emplace_back(app, max_replica_count);
+        }
+    }
+
+    dsn::task_tracker tracker;
+
+    for (auto &task : tasks) {
+        recover_all_partitions_max_replica_count(task.first, task.second, tracker);
+    }
+    tracker.wait_outstanding_tasks();
```

Comment:
How long will it cost, if there is any bug cause the tasks never complete, the meta server will not startup normally, right?